app/models/local_photo.rb

```diff
@@ -600,4 +600,26 @@ def self.delete_images_from_bucket( s3_client, delete_from_bucket, images )
     true
   end
 
+  def self.migrate_to_odp_bucket( start_index, end_index )
+    static_bucket_id = FilePrefix.where( prefix: "https://#{LocalPhoto.s3_host_alias}/photos" ) ).first.id
+    LocalPhoto.joins( "LEFT JOIN flags ON (photos.id = flags.flaggable_id)" ).
+    where( "flags.id IS NULL" ).
+    where( "photos.file_prefix_id=?", static_bucket_id ).
+    where( "photos.license not in (?)", Shared::LicenseModule::LICENSE_NUMBERS - Shared::LicenseModule::ODP_LICENSES ).
+    where( "photos.id between ? and ?", start_index, end_index ).
+    each do |photo|
+      LocalPhoto.change_photo_bucket_and_index_taxa_if_needed( photo )
+    end
+  end
+
+  def self.change_photo_bucket_and_index_taxa_if_needed( p )
+    return unless p = LocalPhoto.find_by_id( p ) unless p.is_a?( LocalPhoto )
+    p.change_photo_bucket_if_needed
+    TaxonPhoto.where( photo_id: p.id ).each do |taxon_photo| 
```

Comment:
Can you extract this into its own instance method - `reindex_taxa_later` or something like that? Then instead of putting it here, have it get called in `self.move_to_appropriate_bucket` near `photo.mark_observations_as_updated`. Then I think this method can be removed and you can call the existing `change_photo_bucket_if_needed` instead of `change_photo_bucket_and_index_taxa_if_needed`. That way taxa will get reindex after any move in the future, not just these related to `migrate_to_odp_bucket`